system/lib/wasmfs/syscalls.cpp

```diff
@@ -426,4 +426,18 @@ __wasi_errno_t __wasi_fd_seek(__wasi_fd_t fd,
 
   return __WASI_ERRNO_SUCCESS;
 }
+
+__wasi_errno_t __wasi_fd_fdstat_get(__wasi_fd_t fd, __wasi_fdstat_t* stat) {
+  auto openFile = wasmFS.getLockedFileTable()[fd];
+  if (!openFile) {
+    return __WASI_ERRNO_BADF;
+  }
+
+  if (openFile.locked().getFile()->is<Directory>()) {
```

Comment:
Should we also set `fs_filetype` to `__WASI_FILETYPE_CHARACTER_DEVICE` when `fd` is `STDOUT_FILENO` or `STDERR_FILENO`? See for example commit https://github.com/emscripten-core/emscripten/pull/13007/commits/530bb9f4c087c067050dd64df481ac8441ec6975 within PR #13007.

Reply:
I think I will leave that up to @ethanalee who is working on wasmfs right now.  This code here is just enough to get the tests passing.  I imagine he will want to report `__WASI_FILETYPE_CHARACTER_DEVICE` based on some property of the getFile() result, yes, but I think that can probably be a followup.

Reply:
Can you add a TODO comment about eventually filling out the other fields of the `stat` struct?

Reply:
Done